test/conformance/serial/ptp.go

```diff
@@ -838,6 +840,260 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
+
+					// Find matches
+					clockClassMap := map[string]bool{"ptp4l": false}
+
+					scanner := bufio.NewScanner(strings.NewReader(buf.String()))
+					timeout := 1 * time.Minute
+					start := time.Now()
+					for scanner.Scan() {
+						t := time.Now()
+						elapsed := t.Sub(start)
+						if elapsed > timeout {
+							fmt.Println("Timedout reading input from metrics")
+						}
+						line := scanner.Text()
+						if matches := clockClassRe.FindStringSubmatch(line); matches != nil {
+							if _, ok := clockClassMap[matches[2]]; ok && matches[3] == "6" {
+								clockClassMap[matches[2]] = true
+								break
+							}
+
+						}
+					}
+					if err := scanner.Err(); err != nil {
+						fmt.Println("Error reading input:", err)
+					}
+					Expect(clockClassMap["ptp4l"]).To(BeTrue(), "Expected ptp4l clock class state to be Locked for GM")
+				})
+
+				By("checking DPLL frequency state locked", func() {
+					/*
+						# TYPE openshift_ptp_frequency_status gauge
+						# openshift_ptp_frequency_status{from="dpll",iface="ens7fx",node="cnfdg32.ptp.eng.rdu2.dc.redhat.com",process="dpll"} 3
+					*/
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn3Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpFrequencyStatus),
+						"frequency status metrics are not detected")
+
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					freqStatusPattern := `openshift_ptp_frequency_status\{from="([^"]+)",iface="([^"]+)",node="([^"]+)",process="([^"]+)"\} (\d+)`
+
+					// Compile the regular expression
+					freqStatusRe := regexp.MustCompile(freqStatusPattern)
+
+					// Find matches
+					freqStatusMap := map[string]bool{"dpll": false}
+
+					scanner := bufio.NewScanner(strings.NewReader(buf.String()))
+					timeout := 1 * time.Minute
+					start := time.Now()
+					for scanner.Scan() {
+						t := time.Now()
+						elapsed := t.Sub(start)
+						if elapsed > timeout {
+							fmt.Println("Timedout reading input from metrics")
+						}
+						line := scanner.Text()
+						if matches := freqStatusRe.FindStringSubmatch(line); matches != nil {
+							if _, ok := freqStatusMap[matches[4]]; ok && matches[5] == "3" {
+								freqStatusMap[matches[4]] = true
+								break
+							}
+
+						}
+					}
+					if err := scanner.Err(); err != nil {
+						fmt.Println("Error reading input:", err)
+					}
+					Expect(freqStatusMap["dpll"]).To(BeTrue(), "Expected dpll frequency status to be Locked for GM")
+
+				})
+
+				By("checking DPLL phase state locked", func() {
+					/*
+						# TYPE openshift_ptp_phase_status gauge
+						# openshift_ptp_phase_status{from="dpll",iface="ens7fx",node="cnfdg32.ptp.eng.rdu2.dc.redhat.com",process="dpll"} 3
```

Comment:
implement checking for multiple cards or add a TODO

Reply:
added a TODO